### PR TITLE
ARROW-592: Add Arrow C++ and GLib DEB and RPM packages

### PR DESCRIPTION
**Pull request by kou**
This just builds .deb and .rpm for the following platforms:

* Debian GNU/Linux stretch amd64
* Ubuntu 16.04 amd64
* Ubuntu 17.04 i386
* Ubuntu 17.04 amd64
* CentOS 6 x86_64
* CentOS 7 x86_64

This doesn't provide APT/Yum repository because we don't have PGP key to sign .deb and .rpm and we don't decide where we put .deb and .rpm files.

Here are .deb and .rpm files built by this: https://packages.groonga.org/tmp/arrow/

### CODE REVIEW

File: cpp-linux/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*.tar.gz
```
Comment (xhochy) [inline cpp-linux/.gitignore:18]: Apache license header

File: cpp-linux/debian/control [hunk 2]
```diff
@@ -0,0 +1,109 @@
+Source: apache-arrow
```
Comment (xhochy) [inline cpp-linux/debian/control:1]: License header

**Review by xhochy — COMMENTED** [review]
LGTM except that not all files have an Apache header. We don't need to be as strict on this as with files in the main repository as most files will not be shipped but it would be nice to add them to all files that can have comments

**Comment by kou** [comment]
For .deb package, there is [`debian/copyright`](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/) file to specify license. It covers `debian/` directory like the following:

https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#example-files-paragraph

> ```text
> Files: debian/*
> Copyright: 2010 Daniela Debianizer
> License: GPL-2+
> ```

Can we use `debian/copyright` instead of putting license header to each file?
I don't know whether all `debian/*` files have "comment syntax". If "comment syntax" doesn't exist, we can't put license header.

File: cpp-linux/.gitignore (hunk 1 above)
Comment (kou) [reply cpp-linux/.gitignore:18]: Done.

**Pull request merged**